Subject: [PATCH] prevent auto combo after raise
---
Index: XIVSlothCombo/AutoRotation/AutoRotationController.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/XIVSlothCombo/AutoRotation/AutoRotationController.cs b/XIVSlothCombo/AutoRotation/AutoRotationController.cs
--- a/XIVSlothCombo/AutoRotation/AutoRotationController.cs	(revision f731ab8c36d8ad8dc4ccac182311df05383dfc2d)
+++ b/XIVSlothCombo/AutoRotation/AutoRotationController.cs	(date 1733040749501)
@@ -29,8 +29,8 @@
         internal static void Run()
         {
             var cfg = Service.Configuration.RotationConfig;
-
-            if (!cfg.Enabled || !Player.Available || Svc.Condition[ConditionFlag.Mounted])
+            
+            if (!cfg.Enabled || !Player.Available || Svc.Condition[ConditionFlag.Mounted] || Player.Status.Any(it => it.SourceId == 418))
                 return;
 
             if (Player.Object.CurrentCastTime > 0) return;
