Subject: [PATCH] toggled with color
---
Index: WrathCombo/Core/Presets.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WrathCombo/Core/Presets.cs b/WrathCombo/Core/Presets.cs
--- a/WrathCombo/Core/Presets.cs	(revision 5cb4ac3c11e7434a26463fa3a473510f900625b1)
+++ b/WrathCombo/Core/Presets.cs	(date 1739440385980)
@@ -5,6 +5,9 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using Dalamud.Game.Text;
+using Dalamud.Game.Text.SeStringHandling;
+using ECommons.ChatMethods;
 using WrathCombo.Attributes;
 using WrathCombo.Combos;
 using WrathCombo.Services;
@@ -229,12 +232,26 @@
             if (!Service.Configuration.EnabledActions.Remove(preset))
             {
                 var ret = EnablePreset(preset);
-                if (outputLog)
-                    DuoLog.Information($"{(int)preset} - {preset} SET{ctrlText}");
+                if (true)
+                    Svc.Chat.Print(new XivChatEntry
+                    {
+                        Message = new SeStringBuilder()
+                            .AddText($"{preset} ")
+                            .AddUiForeground("SET", (ushort)UIColor.LightBlue)
+                            .AddText($"{ctrlText}")
+                            .Build()
+                    });
             }
-            else if (outputLog)
+            else if (true)
             {
-                DuoLog.Information($"{(int)preset} - {preset} UNSET{ctrlText}");
+                Svc.Chat.Print(new XivChatEntry
+                {
+                    Message = new SeStringBuilder()
+                        .AddText($"{preset} ")
+                        .AddUiForeground("UNSET", (ushort)UIColor.Orange)
+                        .AddText($"{ctrlText}")
+                        .Build()
+                });
             }
             return false;
         }
Index: WrathCombo/Commands.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WrathCombo/Commands.cs b/WrathCombo/Commands.cs
--- a/WrathCombo/Commands.cs	(revision 5cb4ac3c11e7434a26463fa3a473510f900625b1)
+++ b/WrathCombo/Commands.cs	(date 1739440468166)
@@ -5,7 +5,6 @@
 using ECommons;
 using ECommons.DalamudServices;
 using ECommons.GameFunctions;
-using ECommons.GameHelpers;
 using ECommons.Logging;
 using Lumina.Excel.Sheets;
 using WrathCombo.Combos;
@@ -195,13 +194,6 @@
         {
             var usablePreset = (CustomComboPreset)preset!;
             method(usablePreset, false);
-
-            var ctrlText = P.UIHelper.PresetControlled(usablePreset) is not null
-                ? " " + OptionControlledByIPC
-                : "";
-
-            DuoLog.Information(
-                $"{target} {(action == toggle ? "toggled" : action)} {ctrlText}");
         }
     }
 
