Subject: [PATCH] ui toggle preset
---
Index: WrathCombo/Window/Functions/Presets.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WrathCombo/Window/Functions/Presets.cs b/WrathCombo/Window/Functions/Presets.cs
--- a/WrathCombo/Window/Functions/Presets.cs	(revision c1a7d2921be2c3f884f8cce8cf00471ee88b68d7)
+++ b/WrathCombo/Window/Functions/Presets.cs	(date 1739440913055)
@@ -108,7 +108,9 @@
             if (info.Name.Contains(" - AoE") || info.Name.Contains(" - Sin"))
                 if (P.UIHelper.PresetControlled(preset) is not null)
                     P.UIHelper.ShowIPCControlledIndicatorIfNeeded(preset);
-
+            
+            if (ImGui.Button($"/wrath toggle {preset}")) ImGui.SetClipboardText($"/wrath toggle {preset}");
+            
             if (P.UIHelper.ShowIPCControlledCheckboxIfNeeded
                     ($"{info.Name}###{preset}", ref enabled, preset, true))
             {
